Core/include/Acts/TrackFitting/KalmanFitter.hpp

```diff
@@ -1150,8 +1150,6 @@ class KalmanFitter {
       track.covariance() = params.covariance().value();
       track.setReferenceSurface(params.referenceSurface().getSharedPtr());
     }
-    track.nMeasurements() = kalmanResult.measurementStates;
-    track.nHoles() = kalmanResult.measurementHoles;
```

Comment:
Aren't `measurementStates` and `measurementHoles` not still redundantly calculated in `propagate`?  Is there a way to not do the calculation both in `propagate` and `calculateTrackQuantities`?

Reply:
We also calculate things like degrees of freedom, total number of states, number of outliers, shared hits in `calculateTrackQuantities`, so we need to iterate over the track states regardless. Technically, both here and in the CKF we have number of measurements, available and don't have to recalculate.

But since we have to loop anyway, I don't see it as a huge problem to do a number of sums there.